src/modules/Api.js

```diff
@@ -0,0 +1,45 @@
+export function fetchAndRenderData() {
+  document.addEventListener('DOMContentLoaded', () => {
+    // Fetch data from the TVmaze API
+    fetch('https://api.tvmaze.com/schedule?country=US&date=2014-12-01')
+      .then((response) => response.json())
+      .then((data) => {
+        const movieList = document.querySelector('.movie-list');
+
+        data.forEach((show) => {
+          const movieDiv = document.createElement('div');
+          movieDiv.className = 'movie';
+
+          const img = document.createElement('img');
+          img.src = show.show.image ? show.show.image.medium : 'placeholder.jpg';
+          img.alt = show.show.name;
+
+          const h2 = document.createElement('h2');
+          const maxLength = 30; // Set a maximum length for the movie name (adjust as needed)
+
+          if (show.show.name.length > maxLength) {
+            h2.textContent = `${show.show.name.substring(0, maxLength)}...`;
+          } else {
+            h2.textContent = show.show.name;
+          }
+
+          const buttonsDiv = document.createElement('div');
+          buttonsDiv.className = 'buttons';
+
+          const likeButton = document.createElement('button');
+          likeButton.className = 'like-button';
+          likeButton.textContent = 'Like';
```

Comment:
Hi @DevKaranJ 

Can you please kindly change the Like-button to a Reservations-button as per the task requirement?
Overall Good job implementing the project as per the requirements 💯

Reply:
Actually it's not functional right now I am hoping to make it work on my next future update but thanks for the review I'll work on it !